From 9927b7dac51cd6ec6a4e23e369f091f578cdd6b2 Mon Sep 17 00:00:00 2001
Message-Id: <9927b7dac51cd6ec6a4e23e369f091f578cdd6b2.1434670190.git.chang-joon.lee@intel.com>
In-Reply-To: <5c49cb8f884ecf227e44159a5a4931d2c0814d5c.1434670190.git.chang-joon.lee@intel.com>
References: <5c49cb8f884ecf227e44159a5a4931d2c0814d5c.1434670190.git.chang-joon.lee@intel.com>
From: vandita kulkarni <vandita.kulkarni@intel.com>
Date: Tue, 2 Jun 2015 00:41:04 +0530
Subject: [PATCH 5/6] REVERTME [VPG]: drm/i915: pipe disable needs pipe/plane
 status to be updated.

The previous pipe/plane status is restored with the present pipe/plane status
at the end of the set_display atomic flip call. On disabling pipe, i.e
during suspend, the planes and pipes are disabled. Here pipe/plane status is
updated, but prev pipe/plane status is not updated. Hence doing the same in
pipe disable case.

REVERTME: This patch is revertme, since it based on the VPG set_display
atomic function which is a revertme patch.

Issue: IMINAN-33414
Change-Id: Ied836a204b1cbe094d8ecfec6d3646993c93a4fd
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 70a7dbc..87c2c4a 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5766,6 +5766,7 @@ static void i9xx_crtc_disable(struct drm_crtc *crtc)
 
 	intel_disable_pipe(dev_priv, pipe);
 	dev_priv->pipe_plane_stat &= ~PIPE_ENABLE(intel_crtc->pipe);
+	dev_priv->prev_pipe_plane_stat = dev_priv->pipe_plane_stat;
 
 	i9xx_pfit_disable(intel_crtc);
 
-- 
1.9.1

